src/BenchmarkDotNet/Portability/Cpu/HardwareIntrinsics.cs

```diff
@@ -52,6 +54,12 @@ static IEnumerable<string> GetCurrentProcessInstructionSets(Platform platform)
                 {
                     case Platform.X86:
                     case Platform.X64:
+                        if (IsX86Avx512FSupported) yield return "AVX-512F";
+                        if (IsX86Avx512BWSupported) yield return "AVX-512BW";
+                        if (IsX86Avx512CDSupported) yield return "AVX-512CD";
+                        if (IsX86Avx512DQSupported) yield return "AVX-512DQ";
+                        if (IsX86Avx512VbmiSupported) yield return "AVX-512VBMI";
+
```

Comment:
The other main path is an `if/else if` pattern.

This new code will yield/return several AVX-512F ISAs and not in their reverse hierarchical order.

Is that intentional?

--------

I'd imagine we want to cover `Avx512F` in the main if/else chain and the others to be on top to be "inline" with how the other works.

Notably certain combinations also have well-defined names.

That is, `x86-64-v4` is the formal definition that includes `AVX512F, AVX512BW, AVX512CD, AVX512DQ, AVX512VL`

`x86-64-v3` is `AVX, AVX2, BMI1, BMI2, F16C, FMA, LZCNT, MOVBE, OSXSAVE`

`x86-64-v2` is `CMPXCHG16B, POPCNT, SSE3, SSE4.1, SSE4.2, SSSE3`

`x86-64-v1` (baseline) is `CMOV, CX8, x87FPU, FXSR, MMX, OSFXSR, SCE, SSE, SSE2`

Reply:
I probably argue against using `x86-x64-v?` since it not immediately obvious what's what.

Instead perhaps we could list it as `AVX-512(X/Y/Z)` e.g. `AVX-512(F/BW/CD/DQ/VL)`. This makes set obvious and immediately visible.